src/cdomains/pthreadDomain.ml

```diff
@@ -0,0 +1,96 @@
+open Prelude
+
+(** Thread ID *)
+module Tid = IntDomain.Flattened
+
+(** Context hash for function calls *)
+module Ctx = IntDomain.Flattened
```

Comment:
Identifying contexts by hash isn't really sound due to hash collisions. This is why witness generation uses `tag` from hashconsing instead.

But I guess it's not worth fixing here because other extract analyses seem to do the same.

Reply:
Is it? I thought it's just potentially less precise since contexts would not be kept apart but thrown together.
But yes, tags or this `ArincUtil.SymTbl` would be better.

Reply:
I guess it might be fine, I'm not familiar with how the extraction things works.

I was just thinking of a case where a hash is used with `Hashtbl.replace` or similar. Then it replaces (which may cause unsoundness by overwriting something) instead of joining values together.

Reply:
I didn't check how these values get used, so you're probably right that it's sound but in a rare case might just be less precise.

Seeing these extraction mechanisms now, at some point it might be worth generalizing and combining these extractions and witness generation, which both want to do this sort of thing.